sips/utils/plotting.py

```diff
@@ -34,157 +35,219 @@ def _asarray(arr: npt.ArrayLike) -> npt.NDArray[np.float32]:
 
-    # Plot points
-    ax.invert_yaxis()
-    for kp_uv, color in zip(keypoints_uv, colors):
-        kp_uv_flat = kp_uv.reshape(-1, 2)
-        ax.scatter(kp_uv_flat[:, 0], kp_uv_flat[:, 1], color=color, alpha=0.2)
-        if kp_uv.ndim == 3:
-            ax.plot(kp_uv[:, :, 0].T, kp_uv[:, :, 1].T, color=color, alpha=0.2)
-
-    # Plot lines
-    if keypoint_pairs is not None:
-        keypoint_pairs = [_asarray(pair) for pair in keypoint_pairs]
-        lines = np.concatenate(keypoint_pairs, axis=-1)
-        lines = lines.reshape(-1, 2 * len(keypoint_pairs))
+    # Plot points / arcs
+    alpha = 1
+    for points_uv, color in [(target_uv, target_color), (source_uv_proj, source_color)]:
+        points_uv = points_uv.detach().cpu()
+
+        if points_uv.size(0) == 2:
+            # Points
+            ax.scatter(points_uv[1], points_uv[0], color=color, alpha=alpha, s=s)
+
+        elif points_uv.size(1) == 2:
+            # Points
+            ax.scatter(points_uv[:, 1], points_uv[:, 0], color=color, alpha=alpha, s=s)
+            # Connect with line
+            lines = points_uv.flatten(2).movedim(0, 1)
+            ax.plot(lines[1], lines[0], color=color, alpha=alpha)
+
+    # Plot keypoint matches
+    if matches is not None:
+        assert len(matches) == 2 and matches[0].size(1) == matches[1].size(1)
+        lines = torch.cat(list(matches), dim=-1).flatten(0, -2).detach().cpu()
         for points in lines:
-            points = points.reshape(len(keypoint_pairs), 2)
-            ax.plot(*points.T, keypoint_pairs_color)
+            points = points.reshape(2, 2).T
+            ax.plot(points[1], points[0], color=match_color)
 
+    ax.set_xlim(-cell_size / 2, width + cell_size / 2)
+    ax.set_ylim(-cell_size / 2, height + cell_size / 2)
+    ax.set_aspect("equal")
+    ax.invert_yaxis()
     fig.tight_layout()
+
     return ax
 
 
 def plot_arcs_3d(
-    keypoints_xyz: Sequence[npt.ArrayLike],
-    camera_pos: Sequence[npt.ArrayLike | None] | None = None,
-    colors: str | Sequence[str] | None = None,
+    projection_arcs: Sequence[torch.Tensor],
+    camera_positions: Sequence[torch.Tensor | None],
+    *,
+    colors: Sequence[str] | None = None,
 ) -> plt.Axes:
     """
     Plot projection arcs in Euclidean (xyz) space.
 
+    Notes
+    -----
+    The default colors are configured for three sonars.
+    They depict the target sonar, the source sonar, and the arcs of the source sonar
+    which are visible in the target sonar, respectively.
+
     Parameters
     ----------
-    keypoints_xyz : Sequence[npt.ArrayLike]
-        Keypoints of shape (..., n_elevations, 3).
-    camera_pos : Sequence[npt.ArrayLike | None] | None, optional
-        Camera position for the (set of) keypoints, by default None
-    colors : str | Sequence[str] | None, optional
-        Colors of the keypoints, by default None
+    projection_arcs : Sequence[torch.Tensor]
+        Projection arcs to be plotted. Shape: (N_ELEVATIONS,3,...)
```

Comment:
Emphasize that this is a Sequence
```suggestion
        Projection arcs to be plotted. Shape: [(N_ELEVATIONS,3,...), ...]
```

Reply:
Can do that